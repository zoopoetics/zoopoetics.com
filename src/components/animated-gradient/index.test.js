import React from 'react';
import renderer from 'react-test-renderer';
import AnimatedGradient from './index';

jest.mock('utils/array-util', () => ({
  getRandomIndex: () => 0,
  rotate: () => [],
  toValueString: () => '',
  wrap: () => [],
}));

describe('AnimatedGradient', () => {
  let component;
  let instance;

  beforeEach(() => {
    component = renderer.create(<AnimatedGradient />);
    instance = component.getInstance();
  });

  it('matches snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('has a getFocusValues function that', () => {
    it('returns an array of random percentages with the passed length', () => {
      const focusValues = instance.getFocusValues(10);
      expect(focusValues.length).toEqual(10);
      expect(focusValues).toEqual(expect.any(Array));
    });
  });
});
