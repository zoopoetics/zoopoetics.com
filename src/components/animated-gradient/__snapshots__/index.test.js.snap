// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedGradient matches snapshot 1`] = `
<svg
  className="sc-bdVaJa cQyftp"
>
  <defs>
    <radialGradient
      fx={0}
      fy={0}
      id="sc-bdVaJa--gradient1"
      r="100%"
    >
      <animate
        attributeName="fx"
        attributeType="XML"
        dur="20s"
        repeatCount="indefinite"
        values=""
      />
      <animate
        attributeName="fy"
        attributeType="XML"
        dur="20s"
        repeatCount="indefinite"
        values=""
      />
      <stop
        offset={0}
        stopColor="rgb(26, 128, 130)"
      >
        <animate
          attributeName="stop-color"
          attributeType="XML"
          dur="20s"
          repeatCount="indefinite"
          values=""
        />
      </stop>
      <stop
        offset={1}
        stopColor="rgb(26, 128, 130)"
      >
        <animate
          attributeName="stop-color"
          attributeType="XML"
          dur="20s"
          repeatCount="indefinite"
          values=""
        />
      </stop>
    </radialGradient>
    <radialGradient
      fx={0}
      fy={0}
      id="sc-bdVaJa--gradient2"
      r="100%"
    >
      <animate
        attributeName="fx"
        attributeType="XML"
        dur="20s"
        repeatCount="indefinite"
        values=""
      />
      <animate
        attributeName="fy"
        attributeType="XML"
        dur="20s"
        repeatCount="indefinite"
        values=""
      />
      <stop
        offset={0}
        stopColor="white"
        stopOpacity={0.3333333333333333}
      />
      <stop
        offset={1}
        stopColor="white"
        stopOpacity={0}
      />
    </radialGradient>
  </defs>
  <rect
    fill="url(#sc-bdVaJa--gradient1)"
    height="100%"
    width="100%"
  />
  <rect
    fill="url(#sc-bdVaJa--gradient2)"
    height="100%"
    width="100%"
  />
</svg>
`;
